pyuvdata/uvdata/mwa_corr_fits.py

```diff
@@ -492,19 +506,26 @@ def _read_fits_file(
                     hdu.data[:, 0::2] + 1j * hdu.data[:, 1::2]
                 )
                 # fill nsample and flag arrays
+                # think about using the mwax weights array in the future
```

Comment:
I believe it is the former. This is from the [docs](https://wiki.mwatelescope.org/display/MP/MWAX+Visibility+File+Format#MWAXVisibilityFileFormat-FITSExtensionWeightsImgHDU):
```
Weights are provided in the gpubox files after each visibility HDU containing one timestep/integration. The weights and how they are determined is discussed below:

    Each visibility has a multiplicative weight applied, based on a data occupancy metric that takes account of any input data blocks that are missing due to lost UDP packets or RFI excision (a potential future enhancement).  The centre (DC) ultrafine channel is excluded when averaging and the centre output channel values are re-scaled accordingly.  Note that only 200 Hz of bandwidth is lost in this process, rather than a complete output channel. 
    As part of the M&C system, the application of weights can be turned on or off per observation based on the science case/needs. With weights not applied, the data will be averaged in the correlator without taking into account the weights. Either way the weights are supplied in each alternate ImgHDU for your information.
```

Reply:
great. Then I think we should just pipe this into the n_sample array. I'm curious about the details of the center channel, I don't quite get that comment. I guess if the application of weights is turned off we should set the n_samples to 1 because everything is included?

Reply:
I think that makes sense.

Reply:
I wonder actually if we would want to keep the weights even if the application is turned off, in case we want to applyy them?

Reply:
I guess I'm not sure, it depends on what is actually being done. I think we should parse this together.

Reply:
Since the proper weights values aren't being provided from MWAX yet, we'll wait to implement them in the reader.